scala/scala.bzl

```diff
@@ -728,4 +818,24 @@ def scala_library_suite(name,
         ts.append(n)
     scala_library(name = name, deps = ts, exports = exports + ts, visibility = visibility)
 
+scala_junit_test = rule(
+  implementation=_scala_junit_test_impl,
+  attrs={
+      "patterns": attr.string_list(default=["Test"]),
+      "_junit": attr.label(default=Label("//external:io_bazel_rules_scala/dependency/junit/junit"), single_file=True),
+      "_hamcrest": attr.label(default=Label("//external:io_bazel_rules_scala/dependency/hamcrest/hamcrest_core"), single_file=True),
+      "_suite": attr.label(default=Label("//src/java/io/bazel/rulesscala/test_discovery:test_discovery")),
+      } + _implicit_deps + _common_attrs,
+  outputs={
+      "jar": "%{name}.jar",
+      "deploy_jar": "%{name}_deploy.jar",
```

Comment:
Because of the code reuse of `scala_binary_common`. `scala_test` does the same.
We can change the common one to optionally generate it and then change both test rules to not generate them. WDYT?
In any case feels like a different PR.